cloud/services/agentpools/agentpools.go

```diff
@@ -87,18 +89,20 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 		// Normalize individual agent pools to diff in case we need to update
 		existingProfile := containerservice.AgentPool{
 			ManagedClusterAgentPoolProfileProperties: &containerservice.ManagedClusterAgentPoolProfileProperties{
-				VMSize:              existingPool.ManagedClusterAgentPoolProfileProperties.VMSize,
-				OsType:              containerservice.Linux,
-				OsDiskSizeGB:        existingPool.ManagedClusterAgentPoolProfileProperties.OsDiskSizeGB,
-				Count:               existingPool.ManagedClusterAgentPoolProfileProperties.Count,
-				Type:                containerservice.VirtualMachineScaleSets,
-				OrchestratorVersion: existingPool.ManagedClusterAgentPoolProfileProperties.OrchestratorVersion,
-				VnetSubnetID:        existingPool.ManagedClusterAgentPoolProfileProperties.VnetSubnetID,
+				Count:               existingPool.Count,
+				OrchestratorVersion: existingPool.OrchestratorVersion,
+			},
+		}
+		// Normalized profile to check for update.
+		normalizedProfile := containerservice.AgentPool{
+			ManagedClusterAgentPoolProfileProperties: &containerservice.ManagedClusterAgentPoolProfileProperties{
+				Count:               profile.Count,
+				OrchestratorVersion: profile.OrchestratorVersion,
 			},
 		}
 
 		// Diff and check if we require an update
-		diff := cmp.Diff(profile, existingProfile)
+		diff := cmp.Diff(normalizedProfile, existingProfile)
```

Comment:
does this mean all other fields (including az) can not be changed?

Reply:
yes it cannot be mutable.